File: validateSAID.vb
' Dont forget to import System.Text.RegularExpressions
Public Function check(id As String) As Integer
	id = id.Trim()
	If Not Regex.IsMatch(id, "[0-9]+") Then
		Return -1
	End If
	If id.Length <> 10 Then
		Return -1
	End If
	Dim type As Integer = Integer.Parse(id.Chars(0))
	If type <> 2 AndAlso type <> 1 Then
		Return -1
	End If
	Dim sum As Integer = 0
	For i As Integer = 0 To 9
		If i Mod 2 = 0 Then
			Dim ZFOdd As String = (Integer.Parse(id.Chars(i)) * 2).ToString().PadLeft(2, "0")
			sum += Integer.Parse(ZFOdd.Chars(0)) + Integer.Parse(ZFOdd.Chars(1))
		Else
			sum += Integer.Parse(id.Chars(i))
		End If
	Next
	Return If((sum Mod 10 <> 0), -1, type)
	
End Function

File: validateSAID.cs
class ValidateSAID {

    public int check(String id) {
        id = id.Trim();
        if (!Regex.IsMatch(id, @"[0-9]+")) {
            return -1;
        }
        if (id.Length != 10) {
            return -1;
        }
        int type = (int)(id[0]-'0');
        if (type != 2 && type != 1) {
            return -1;
        }
        int sum = 0;
        for (int i = 0; i < 10; i++) {
            if (i % 2 == 0) {
                String ZFOdd = ((int)(id[i]-'0') * 2).ToString().PadLeft(2, '0');
                sum += (int)(ZFOdd[0]-'0') + (int)(ZFOdd[1]-'0');
            } else {
                sum += (int)(id[i]-'0');
            }
        }
        return (sum % 10 != 0) ? -1 : type;

    }

} 
